examples/ssd1306-i2c.rs

```diff
@@ -0,0 +1,191 @@
+#![no_std]
+#![no_main]
+
+extern crate esp32_hal as hal;
+extern crate xtensa_lx6_rt;
+
+use {
+    core::{fmt::Write, panic::PanicInfo},
+    hal::{
+        clock_control::{self, CPUSource, ClockControl, ClockControlConfig},
+        dport::Split,
+        dprintln, i2c,
+        prelude::*,
+        serial::{config::Config as SerialConfig, NoRx, NoTx, Serial},
+        target,
+    },
+    ssd1306::{prelude::*, Builder},
+    xtensa_lx6_rt::get_cycle_count,
+};
+
+const CORE_HZ: u32 = 10_000_000;
+
+const WDT_WKEY_VALUE: u32 = 0x50D83AA1;
+
+#[no_mangle]
+fn main() -> ! {
+    let dp = unsafe { hal::target::Peripherals::steal() };
+
+    let mut timg0 = dp.TIMG0;
+    let mut timg1 = dp.TIMG1;
+    let (mut dport, dport_clock_control) = dp.DPORT.split();
+
+    // (https://github.com/espressif/openocd-esp32/blob/97ba3a6bb9eaa898d91df923bbedddfeaaaf28c9/src/target/esp32.c#L431)
+    // openocd disables the wdt's on halt
+    // we will do it manually on startup
+    disable_timg_wdts(&mut timg0, &mut timg1);
+
+    // setup clocks & watchdog
+    let mut clkcntrl = ClockControl::new(
+        dp.RTCCNTL,
+        dp.APB_CTRL,
+        dport_clock_control,
+        clock_control::XTAL_FREQUENCY_40M,
+    )
+    .unwrap();
+
+    let (clkcntrl_config, mut watchdog) = clkcntrl.freeze().unwrap();
+    watchdog.disable();
+
+    let pins = dp.GPIO.split();
+
+    let mut rst = pins.gpio16.into_push_pull_output();
+
+    //i2c_set_pin(0, 4, 15, true, true, i2c_mode_t::I2C_MODE_MASTER);
+    let gpio = unsafe { &*target::GPIO::ptr() };
+    let iomux = unsafe { &*target::IO_MUX::ptr() };
+
+    let sda = pins.gpio4.into_alternate_3();
+    let scl = pins.gpio15.into_alternate_3();
+
+    // sda
+    {
+        // gpio_set_level(sda_io_num, I2C_IO_INIT_LEVEL);
+        gpio.pin4.write(|w| unsafe { w.bits(1 << 4) });
```

Comment:
This is wrong. The pin4 register is not used to set the output level, but a.o. the pad_driver mode.

(And this is not set correctly in the gpio.rs file: but this is not solving the problem.)